src/team/data_preprocessing.py

```diff
@@ -75,35 +77,36 @@ def select_reference_index(self) -> int:
         cumulative_dist = np.sum(ds, axis=1)
         return np.argwhere(cumulative_dist == np.min(cumulative_dist))[0][0]
 
-    def _align_data(self) -> None:
+    def _align_data(self, window: Optional[float] = None) -> None:
         """
         Aligns demonstrations with respect to the reference one using the dynamic
         time warping algorithm (DTW)
 
+        :param window: only allow for maximal shifts from the two diagonals smaller
+        than this number in seconds. Default to maximum.
         """
         # end effector position information considered
         tcp_ref = self.reference.tcp
         path = None
         for i, trajectory in enumerate(self.trajectories_to_align):
             if i == self.reference_index:
                 continue
-            _, paths = dtw_ndim.warping_paths(tcp_ref, trajectory.tcp)
-            # best matching transformation
-            path = dtw.best_path(paths)
-            aligned_path = [p[1] for p in path]
+
+            if window is not None:
+                # Convert windows in second to measurements.
+                window = int(
+                    window
+                    / max(trajectory.average_sampling, self.reference.average_sampling)
+                )
```

Comment:
Shouldn't be the other way round to convert window from seconds to measurements instead? :)

Reply:
To convert `average_sampling` in seconds I do `1/average_sampling`. Then to get `window` which is here expressed in seconds, I multiply `1/average_sampling` by `window`. To me it is in this order :). I will clarify the documentation. Tell me if you agree.

Reply:
God you are right hahaha. This is why I should code late at night :S

Reply:
Ok this should be solved